src/zkp/mod.rs

```diff
@@ -26,6 +26,29 @@ use merlin::Transcript;
 use rand::{CryptoRng, RngCore};
 use serde::{de::DeserializeOwned, Serialize};
 
+/// A trait for constructing arbitrary system context.
+///
+/// This context can be turned into bytes and appended to a [`Transcript`].
+pub(crate) trait ProofContext {
+    fn as_bytes(&self) -> &[u8];
+}
+
+/// This is a temporary preliminary implementation.
```

Comment:
nit: Can you either link to an issue in which it'll be removed or explain the conditions under which we can remove it? I actually think we can take it out (or `#[cfg(test)]` it) as soon as we do #240.